advisor/src/main/kotlin/advisors/VulnerableCode.kt

```diff
@@ -116,4 +116,18 @@ class VulnerableCode(name: String, vulnerableCodeConfiguration: VulnerableCodeCo
         if (scores.isEmpty()) return listOf(VulnerabilityReference(sourceUri, null, null))
         return scores.map { VulnerabilityReference(sourceUri, it.scoringSystem, it.value) }
     }
+
+    /**
+     * Return a meaningful identifier for this vulnerability that can be used in reports. Obtain this identifier from
```

Comment:
I see a number of use cases for having an "official" vulnerability ID instead of an internal one:
* Reports display a meaningful ID readers are familiar with.
* In the current implementation, the results of different advisors are combined by the vulnerability IDs. This can only work if common IDs are used. That way, it is also easier possible to compare the results of different advisors.
* Resolutions for vulnerabilities match the vulnerability ID. Here it is again easier for users to use an official ID. And if different advisors are enabled that report the same vulnerabilities under different IDs, multiple resolutions have to be written.

It may well be the case that the single vulnerability ID we have currently in the model is not suitable to support all these use cases. One possible solution could be using aliases in a similar way as done by VulnerableCode: The vulnerability ID would then be internal, but a number of other IDs could be assigned. This is a bigger change though. And if we supported multiple aliases, the matching of vulnerabilities could also become complex.

Reply:
Hm,

1. Displaying a meaningful ID could be implemented by displaying aliases
2. The implementation of combining vulnerabilities could be changed to work on aliases. It could be more complex, 
    but it would also then consider all aliases in the de-deduplication. The current de-deduplication with the change 
    as-is would consider only one alias, the one arbitrarily chosen as id. Is that correct? If so, would that not need to  
    get extended so that it becomes similarily complex?
3. Maybe the resolution mechanism should work on aliases in addition to the internal id?
    As-is IIUC correctly if there are multiple aliases the problem would be solved only for the first one right?

Reply:
You are right that this PR does not provide a full solution for all of these points; this is what I wanted to say in the last paragraph of my previous comment. I have updated the commit message.

Reply:
Thanks!